js/hang/src/watch/video/source.ts

```diff
@@ -96,6 +125,29 @@ export class Source {
 		this.#signals.effect(this.#runPending.bind(this));
 		this.#signals.effect(this.#runDisplay.bind(this));
 		this.#signals.effect(this.#runJitter.bind(this));
+
+		this.#signals.effect((effect) => {
+			const currentFrame = effect.get(this.frame);
+			const nextFrame = this.#next;
+			const enabled = effect.get(this.enabled);
+
+			const isBufferEmpty = enabled && !currentFrame && !nextFrame;
+
+			if (isBufferEmpty && this.#bufferStallStartTime === null) {
+				this.#bufferStallStartTime = Date.now();
+				this.bufferStatus.set({ reason: PlayerStatusReason.BUFFER_EMPTY });
+			} else if (enabled && !isBufferEmpty && this.#bufferStallStartTime !== null) {
+				// How long the buffer was empty/stalled
+				const stallDuration: number = Date.now() - this.#bufferStallStartTime;
```

Comment:
What is this used for? Could/should the consumer of the signal keep track of this instead?